azure-functions/src/durable/history.rs

```diff
@@ -17,13 +17,12 @@ pub struct HistoryEvent {
 
     pub(crate) event_id: i32,
 
-    #[serde(alias = "IsPlayed")]
-    pub(crate) played: bool,
+    pub(crate) is_played: bool,
 
     pub(crate) timestamp: DateTime<Utc>,
 
-    #[serde(default, alias = "IsProcessed")]
-    pub(crate) processed: bool,
+    #[serde(skip)]
```

Comment:
Why skip deserializing of this field? Doesn't it lead to an error when we have 2 orchestration event, one of which is processed already?

Reply:
It's not in the incoming history data, just a flag used in the context.

Reply:
Interesting, I remember I saw this field in spec.

Reply:
I could be wrong about this.  I implemented `select_all` last night and saw that replay isn't working entirely correctly when fanning out and selecting on each activity, so I'll double check that's the case, otherwise I'll restore this change.

Reply:
It doesn't appear to ever be passed in the history.  I think it's for bookkeeping only in the JavaScript.

I think the issue with the replay is that my implementation naively moves to the next orchestration started event upon each `Poll::Ready`, which isn't always correct.

For example:

```rust
if !context.is_replaying() {
    warn!("Orchestration started at {}.", context.current_time());
}

let mut activities = vec![
    context.call_activity("say_hello", "Tokyo"),
    context.call_activity("say_hello", "London"),
    context.call_activity("say_hello", "Seattle"),
];

let mut completed = 0;

while !activities.is_empty() {
    let (r, _, remaining) = context.select_all(activities).await;

    completed += 1;

    if !context.is_replaying() {
        warn!("Result #{} is {}.", completed, r.unwrap());
    }

    activities = remaining;
}

if !context.is_replaying() {
    warn!("Orchestration completed at {}.", context.current_time());
}
```

This fans out to three activities and prints the results of each as they complete.

The bug I'm noticing is when the orchestration function is being invoked once for two out of the three activities completing and then again for the remaining one.  Because my implementation just blindly moves forward to the next orchestration started event, it causes the second result to be logged twice: the first time is correct because it is not replaying (both completed events are part of the most recent execution), but the second time is incorrect because it moved "too far" in the history and into the third and final completion, causing it to be treated as not being replayed when it is.

I have a fix in mind that I'll implement tonight and get the `select_all` implementation up for review.